docs/source/1.0/spec/aws/aws-cloudformation.rst

```diff
@@ -0,0 +1,672 @@
+
+.. tabs::
+
+    .. code-tab:: smithy
+
+        namespace smithy.example
+
+        use aws.cloudformation#additionalIdentifier
+        use aws.cloudformation#createOnlyProperty
+        use aws.cloudformation#excludeProperty
+        use aws.cloudformation#mutableProperty
+        use aws.cloudformation#readOnlyProperty
+        use aws.cloudformation#writeOnlyProperty
+        use aws.cloudformation#resource
+
+        @resource(additionalSchemas: [FooProperties])
+        resource Foo {
+            identifiers: {
+                fooId: String,
+            },
+            create: CreateFoo,
+            read: GetFoo,
+            update: UpdateFoo,
+        }
+
+        @http(method: "POST", uri: "/foos", code: 200)
+        operation CreateFoo {
+            input: CreateFooRequest,
+            output: CreateFooResponse,
+        }
+
+        structure CreateFooRequest {
+            @mutableProperty
+            tags: TagList,
+
+            @writeOnlyProperty
+            secret: String,
+
+            fooAlias: String,
+
+            mutableProperty: ComplexProperty,
+            createProperty: ComplexProperty,
+            writeProperty: ComplexProperty,
+        }
+
+        structure CreateFooResponse {
+            fooId: String,
+        }
+
+        @readonly
+        @http(method: "GET", uri: "/foos/{fooId}", code: 200)
+        operation GetFoo {
+            input: GetFooRequest,
+            output: GetFooResponse,
+        }
+
+        structure GetFooRequest {
+            @httpLabel
+            @required
+            fooId: String,
+
+            @httpQuery("fooAlias")
+            @additionalIdentifier
+            fooAlias: String,
+        }
+
+        structure GetFooResponse {
+            fooId: String,
+
+            @httpResponseCode
+            @excludeProperty
+            responseCode: Integer,
+
+            @readOnlyProperty
+            updatedAt: Timestamp,
+
+            mutableProperty: ComplexProperty,
+            createProperty: ComplexProperty,
+            readProperty: ComplexProperty,
+        }
+
+        @idempotent
+        @http(method: "PUT", uri: "/foos/{fooId}", code: 200)
+        operation UpdateFoo {
+            input: UpdateFooRequest,
+        }
+
+        structure UpdateFooRequest {
+            @httpLabel
+            @required
+            fooId: String,
+
+            fooAlias: String,
+            writeProperty: ComplexProperty,
+            mutableProperty: ComplexProperty,
+        }
+
+        structure FooProperties {
+            addedProperty: String,
+
+            @mutableProperty
+            barProperty: String,
+
+            @createOnlyProperty
+            immutableSetting: Boolean,
+
+            @readOnlyProperty
+            createdAt: Timestamp,
+
+            @writeOnlyProperty
+            password: String,
+        }
+
+        structure ComplexProperty {
+            anotherProperty: String,
+        }
+
+The computed resource property mutabilities are:
```

Comment:
It would be nice to first given an example of something with as few traits as possible, and then a more kitchen-sink example like what you have here.